api/v1alpha3/backstage_types.go

```diff
@@ -0,0 +1,323 @@
+	// Otherwise, only the specified key will be mounted as a file.
+	// Bear in mind not to put sensitive data in those ConfigMaps. Instead, your app-config content can reference
+	// environment variables (which you can set with the ExtraEnvs field) and/or include extra files (see the ExtraFiles field).
+	// More details on https://backstage.io/docs/conf/writing/.
+	// +optional
+	ConfigMaps []ObjectKeyRef `json:"configMaps,omitempty"`
+}
+
+type ExtraFiles struct {
+	// Mount path for all extra configuration files listed in the Items field
+	// +optional
+	// +kubebuilder:default=/opt/app-root/src
+	MountPath string `json:"mountPath,omitempty"`
+
+	// List of references to ConfigMaps objects mounted as extra files under the MountPath specified.
+	// For each item in this array, if a key is not specified, it means that all keys in the ConfigMap will be mounted as files.
+	// Otherwise, only the specified key will be mounted as a file.
+	// +optional
+	ConfigMaps []ObjectKeyRef `json:"configMaps,omitempty"`
+
+	// List of references to Secrets objects mounted as extra files under the MountPath specified.
+	// For each item in this array, a key must be specified that will be mounted as a file.
+	// +optional
+	Secrets []ObjectKeyRef `json:"secrets,omitempty"`
+
+	// List of references to Persistent Volume Claim objects mounted as extra files
+	// For each item in this array, a key must be specified that will be mounted as a file.
+	// +optional
+	Pvcs []PvcRef `json:"pvcs,omitempty"`
+}
+
+type ExtraEnvs struct {
+	// List of references to ConfigMaps objects to inject as additional environment variables.
+	// For each item in this array, if a key is not specified, it means that all keys in the ConfigMap will be injected as additional environment variables.
+	// Otherwise, only the specified key will be injected as an additional environment variable.
+	// +optional
+	ConfigMaps []ObjectKeyRef `json:"configMaps,omitempty"`
+
+	// List of references to Secrets objects to inject as additional environment variables.
+	// For each item in this array, if a key is not specified, it means that all keys in the Secret will be injected as additional environment variables.
+	// Otherwise, only the specified key will be injected as environment variable.
+	// +optional
+	Secrets []ObjectKeyRef `json:"secrets,omitempty"`
+
+	// List of name and value pairs to add as environment variables.
+	// +optional
+	Envs []Env `json:"envs,omitempty"`
+}
+
+type ObjectKeyRef struct {
+	// Name of the object
+	// We support only ConfigMaps and Secrets.
+	//+kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Key in the object
+	// +optional
+	Key string `json:"key,omitempty"`
+
+	// Path to mount the Object. If not specified default-path/Name will be used
+	// +optional
+	MountPath string `json:"mountPath"`
+
+	// Whether subPath is used to mount
+	// +optional
+	//+kubebuilder:default=true
+	WithSubPath *bool `json:"withSubPath"`
```

Comment:
Unless I've missed it, this field does not seem to be used. Can it be removed from the CRD to avoid any confusion?

Reply:
I plan to use it for https://issues.redhat.com/browse/RHIDP-4454, added it as part of 4370

Reply:
commented it out for now